plugins/source/aws/codegen/recipes/backup.go

```diff
@@ -23,9 +23,10 @@ func BackupResources() []*Resource {
 			},
 		},
 		{
-			SubService: "plans",
-			Struct:     &backup.GetBackupPlanOutput{},
-			SkipFields: []string{"BackupPlanArn"},
+			SubService:          "plans",
+			Struct:              &backup.GetBackupPlanOutput{},
+			SkipFields:          []string{"BackupPlanArn"},
+			PreResourceResolver: "getPlan",
```

Comment:
Shouldn't the `PreResourceResolver` be a plural, because it is getting a list of items?

Reply:
`Pre` is actually not a good name, because it's the last call before we "realize" a resource with its fields, **after** the actual resolver is called (to potentially get the list of items). The "singularization" of the list of items is done by the SDK, and we get a single handle from the resolver, look it up in the `PreResourceResolver` handler and do a `resource.Item` assignment (or `resource.SetItem()`, which is never used for some reason)